cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -74,20 +75,39 @@ private[lagom] object JoinClusterImpl {
 
     }
 
-    CoordinatedShutdown(system).addTask(CoordinatedShutdown.PhaseClusterShutdown, "exit-jvm-when-downed") {
+    CoordinatedShutdown(system).addTask(PhaseClusterShutdown, "exit-jvm-when-downed") {
       () =>
-        val reasonIsDowning = CoordinatedShutdown(system)
-          .shutdownReason().contains(CoordinatedShutdown.ClusterDowningReason)
+        val shutdownReason: Option[Reason] = CoordinatedShutdown(system).shutdownReason()
 
-        // if 'exitJvm' is enabled and the trigger (aka Reason) for CoordinatedShutdown is ClusterDowning
-        // we must exit the JVM. This can lead to the cluster closing before the `Application` but we're
-        // out of the cluster (downed) already so the impact is acceptable. This will be improved when
-        // Play delegates all shutdown logic to CoordinatedShutdown in Play 2.7.x.
+        val reasons: Seq[Reason] = Seq(
+          ClusterDowningReason,
+          ClusterJoinUnsuccessfulReason,
+          IncompatibleConfigurationDetectedReason)
+
+        val reasonIsDowning: Boolean = shutdownReason.exists(reasons.contains)
+
```

Comment:
The few lines before this point are the actual change. Since Akka 2.5.22, there are multiple cluster-related events that require a node shutdown.